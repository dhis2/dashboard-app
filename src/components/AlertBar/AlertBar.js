import React from 'react'
import { connect } from 'react-redux'
import { AlertBar, AlertStack } from '@dhis2/ui'
import PropTypes from 'prop-types'

import { sGetAlertMessage } from '../../reducers/alert'
import { acClearAlertMessage } from '../../actions/alert'

export const Alert = ({ message, onClose }) =>
    message ? (
        <AlertStack>
            <AlertBar onHidden={onClose} permanent>
                {message}
            </AlertBar>
        </AlertStack>
    ) : null

Alert.propTypes = {
    message: PropTypes.string,
    onClose: PropTypes.func,
}

const mapStateToProps = state => ({
    message: sGetAlertMessage(state),
})

export default connect(mapStateToProps, {
    onClose: acClearAlertMessage,
})(Alert)
