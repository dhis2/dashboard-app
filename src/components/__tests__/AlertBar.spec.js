import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { Alert } from '../AlertBar'

describe('AlertBar', () => {
    it('renders alert message', () => {
        const AlertBar = shallow(
            <Alert message="Luke I am your father" onClose={jest.fn()} />
        )
        expect(toJson(AlertBar)).toMatchSnapshot()
    })

    it('renders nothing when no message', () => {
        const AlertBar = shallow(<Alert />)
        expect(toJson(AlertBar)).toMatchSnapshot()
    })
})
