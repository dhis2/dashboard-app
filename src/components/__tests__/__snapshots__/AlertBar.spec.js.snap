// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertBar renders alert message 1`] = `
<AlertStack
  dataTest="dhis2-uicore-alertstack"
>
  <AlertBar
    dataTest="dhis2-uicore-alertbar"
    duration={8000}
    icon={true}
    onHidden={[MockFunction]}
    permanent={true}
  >
    Luke I am your father
  </AlertBar>
</AlertStack>
`;

exports[`AlertBar renders nothing when no message 1`] = `""`;
